### ISSUE TITLE
The released training code does not implement the anchor-based diffusion action head described in the paper

### ISSUE THREAD

**Issue by Y1ZzLu** [om-ai-lab/OpenTrackVLA#33]
### Description

Thank you for releasing TrackVLA/OmTrackVLA. I am trying to reproduce the trajectory-planning method described in the paper:

[TrackVLA: Embodied Visual Tracking in the Wild](https://arxiv.org/html/2505.23189)

However, the released training code appears to implement a deterministic waypoint regression head rather than the anchor-based diffusion policy described in the paper.

I may be misunderstanding the intended release scope, so I would appreciate clarification.

### Difference between the paper and the released code

According to Section 3.1 and Appendix A/B of the paper, the planning branch should:

1. Collect all training trajectories and apply K-means clustering to obtain trajectory anchors.
2. Add Gaussian noise to the anchors.
3. Use an action model to denoise the noisy anchors conditioned on the LLM output.
4. Predict both:
   - denoised trajectories;
   - trajectory classification scores.
5. Use the tracking loss:

   ```text
   L_track = sum_i [
       s_i * MSE(tau_hat_i, tau_gt)
       + lambda * BCE(s_hat_i, s_i)
   ]

**Comment by kyusonglee** [comment]
We explored using a diffusion transformer for the action head, but couldn't reproduce the quality reported in the original TrackVLA paper. The trajectories didn't look good enough. We switched to an MLP-based approach instead, which proved to work well. The main advantage is its simplicity.
